src/Components/Activity/Activity.js

```diff
@@ -21,47 +20,69 @@ const handleButtonClick = (e) => {
         ApiData.fetchAll()
         .then(data => setActivity(data))
         setOptionValue('Select Category')
+        setDisabled(true)
     }else {
         ApiData.fetchType(optionValue)
         .then(data => setActivity(data))
         setOptionValue('Select Category')
+        setDisabled(true)
     }
+}
 
+const toggleFavorites = (activity) => {
+    if (!favorite) {
+        let favoritesInStorage = JSON.parse(localStorage.getItem('favorites')) || []
+        favoritesInStorage = [...favoritesInStorage, activity]
+        localStorage.setItem('favorites', JSON.stringify(favoritesInStorage))  
+    } else {
+        let favoritesInStorage = JSON.parse(localStorage.getItem('favorites')) || []
+        favoritesInStorage = favoritesInStorage.filter((f) => f.key !== activity.key)
+        localStorage.setItem('favorites', JSON.stringify(favoritesInStorage))
+    }
+    setFavorite(!favorite)
 }
+
     return (
     <div className="Activity-container">
     <nav className="nav-bar">
-        <button>Favorites</button>
+    <Link to='/Activity/Favorites'>
+    <button className="favorites-btn">Favorites </button>
+    </Link>
     </nav>
     <section className="randomizer-container">
-    <div className="dropdown-container">
-    <select className="drop-down" id="dropDown" value={optionValue} onChange={handleCategory}>
-    <option value=""> {optionValue}</option>
-    <option value="all">All</option>
-    <option value="education">Education</option>
-    <option value="recreational">Recreational</option>
-    <option value="social">Social</option>
-    <option value="diy">DIY</option>
-    <option value="busywork">Busywork</option>
-    <option value="music">Music</option>
-    <option value="charity">Charity</option>
-    <option value="cooking">Cooking</option>
-    <option value="relaxation">Relaxation</option>
+        <div className="dropdown-container">
+    <select className="drop-down" 
+    id="dropDown" value={optionValue} 
+    onChange={handleCategory}>
+        <option value=""> {optionValue}</option>
+        <option value="all">All</option>
+        <option value="education">Education</option>
+        <option value="recreational">Recreational</option>
+        <option value="social">Social</option>
+        <option value="diy">DIY</option>
+        <option value="busywork">Busywork</option>
+        <option value="music">Music</option>
+        <option value="charity">Charity</option>
+        <option value="cooking">Cooking</option>
+        <option value="relaxation">Relaxation</option>
     </select>
-    </div>
+        </div>
     <article className="activity-container">
     {activity && <section className="activity-card">
-    <div onClick={() => setFavorite(true)}>{!favorite ? '🤍' : '🧡'}</div>
+    <div className='favorite-btn' onClick={() => toggleFavorites(activity)}>{!favorite ? '🤍' : '🧡'}</div>
```

Comment:
Great logic! Awesome use of changing the image dynamically. And I love the simplification by using emoji's